### PR TITLE
[dart_ui_web_shim] Introduce package.

### PR DESCRIPTION
**Pull request by ditman**
* Package the `dart:ui` web shim so it can be used in multiple places without having multiple copies of the source code (there are at least 4 copies of this shim code between flutter/packages and flutter/plugins).
* Add tests to the shim, that was untested earlier.

## Issues

* Fixes https://github.com/flutter/flutter/issues/41563  (workaround until the code is exposed from a different package than dart:ui)

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the `#hackers-new` channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/dart_ui_web_shim/example/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+class _MyAppState extends State<MyApp> {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        body: Column(
+          children: const <Widget>[
+            SizedBox(
+              child: HtmlElementView(
+                viewType: 'defined-in-tests',
```
Comment (ditman) [inline packages/dart_ui_web_shim/example/lib/main.dart:48]: Use `viewId` here

File: packages/dart_ui_web_shim/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: dart_ui_web_shim
+description: A shim so dart analyze doesn't complain about web-only parts of the dart:ui package.
+repository: https://github.com/flutter/packages/tree/master/packages/dart_ui_web_shim
+issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+dart_ui_web_shim%22
+version: 0.1.0
+
+environment:
+  sdk: ">=2.12.0 <3.0.0"
+  flutter: ">=1.17.0"
```
Comment (stuartmorgan) [inline packages/dart_ui_web_shim/pubspec.yaml:9]: 2.0, presumably.

**Review by stuartmorgan — COMMENTED** [review]
This mostly makes sense to me, but it seems dangerous to shim something that comes from the SDK, which we can't control the version of. What happens if these methods change? Won't everyone with dependency on this package be broken?

**Comment by ditman** [comment]
> What happens if these methods change? Won't everyone with dependency on this package be broken?

Yeah, that's a good point. Let me ask in Flutter Web channel to see if we have a path forward for those. Ideally, we should make the next version backwards compatible by supporting both the old and new APIs, and then we'd have time to fix the package, but I want to make sure.

**Comment by ditman** [comment]
We're closing this, we don't want to make this web-only API's that need to be removed too easy to use, so off this goes for now!

**Pull request closed without merging**